### DISCUSSION TITLE
Prevent popover from flipping position

### DISCUSSION THREAD

**Discussion by viveknigam3003** [mantinedev/mantine#4184] [Feature requests]
I know in most cases it might be beneficial to flip the position for making the popover visible always, but in some cases, it could be useful to give an option to prevent the flipping.

I am using the Popover through the HoverCard component and in the Date-Time Picker component of v6.x.x and the whole page is going unresponsive when there is not enough space below it or the popover needs to be rendered near the edge of the screen because the position is being calculated constantly. Now this could be due to my project's setup as well, but since the codebase I'm working with is huge, it's extremely difficult to figure out what is conflicting with this calculation.

See in the video, that when I'm hovering over the last user avatar (near the edge) it is going in an infinite loop.

https://user-images.githubusercontent.com/30192068/235942205-16e84d5d-3eac-4265-a659-f1df74bfc21d.mov

This is the code which is printing the values
```tsx
<HoverCard
    shadow="md"
    openDelay={500}
    onPositionChange={pos => {
      console.log('Position changed', pos);
    }}
    position={'bottom'}
>
     <HoverCard.Target>
         {/* User avatar */}
     </HoverCard.Target>
     <HoverCard.Dropdown>
         {/* Box component with content */}
    </HoverCard.Dropdown>
</HoverCard>
```

I'd like to propose an option to disable this auto position and force the position to the one specified in the 'position' prop.

However, it is notable that the **`keepMounted` prop is helping me avoid this problem**, but making my dom heavy with these elements.

<img width="810" alt="Screenshot 2023-05-03 at 6 48 35 PM" src="https://user-images.githubusercontent.com/30192068/235937580-54633ebf-04ac-487d-a580-5c4d454c898d.png">

Let me know if there is some other workaround for it, or else I'll be happy to contribute to adding this prop.

**Comment by viveknigam3003** [comment]
@rtivital any thoughts and ideas on this?

**Reply by rtivital** [reply]
You can configure floating ui with `middlewares` prop:

```tsx
<Popover middlewares={{ flip: false, shift: true, inline: false }} />
```

**Reply by viveknigam3003** [reply]
Yess! Just found that the same after going through the code like 20-25 mins ago. Thanks a lot for the reply! 🙏🏼
